### ISSUE TITLE
model dump crashed

### ISSUE THREAD

**Issue by JerryX2014** [microsoft/LightGBM#2703]
Hello,

I was running lightGBM 2.3.1 on linux environment. When I tried to dump a lighGBM model built a while ago, the system crashed. Below is the error message.

A fatal error has been detected by the Java Runtime Environment:
SIGSEGV (0xb) at pc=0x00007fa4dbba430a, pid=4128, tid=0x00007fa4f5295840
JRE version: Java(TM) SE Runtime Environment (8.0_121-b13) (build 1.8.0_121-b13)
Java VM: Java HotSpot(TM) 64-Bit Server VM (25.121-b13 mixed mode linux-amd64 compressed oops)
Problematic frame:
C  [lib_lightgbm.so+0x6330a]  LGBM_BoosterDumpModel+0x2a

However, if I rerun the modeling script and it worked. Was it because the pointer to the model environment was invalid? How can I check that? 

Thanks!

**Comment by StrikerRUS** [comment]
@JerryX2014 Hi!
Is it possible to provide a code you run?

**Comment by JerryX2014** [comment]
Hi @StrikerRUS,

The code is pretty easy. demo.lgbdata contains all matrices and monotone constraints. You can replace it with other datasets. 

params <- list(objective = "regression", metric = "l2",
               monotone_constraints = paste0(sep=',',demo.lgbdata$monotone,collapse = ''))
demo.lgb_model3 <- lgb.train(
  params = params
  , data = demo.lgbdata$watchlist$train
  , nrounds = 20
  , valids = list(valid=demo.lgbdata$watchlist$eval)
  , min_data = 1
  , learning_rate = 1
  , early_stopping_rounds = 5
)

Below are two screenshots related to the memory pointer. demo.lgb_model is an old model I run a couple of weeks ago, while demo.lgb_model3 is one I just run. demo.lgb_model3 can be dumped successfully, but demo.lgb_model would trigger a crash. I found that the memory locations of .__enclos_env__ environment and private environment for demo.lgb_model were quite difference. For demo.lgb_model3, those two locations are pretty close (definitely). 

![image](https://user-images.githubusercontent.com/10041052/72908376-62ccb300-3d03-11ea-908c-d72b09c5eb8c.png)

![image](https://user-images.githubusercontent.com/10041052/72908738-eab2bd00-3d03-11ea-84d3-b4f60f8c8e8c.png)

Hope these help. Let me know if you need more information.

**Comment by StrikerRUS** [comment]
@JerryX2014 Sorry, can you please explain how code in R language and Java SIGSEGV are connected?

**Comment by StrikerRUS** [comment]
Closing due to the inactivity. Feel free to reopen with more details.
